src/winning_check.c

```diff
@@ -0,0 +1,258 @@
+        player_score[i]=0;
+    }
+    int highest_cart_value[numbers_of_player];//highest cart in players hand// include table carts important in looking for winer without any set
+    int start_of_sequence[numbers_of_player];//cart that start sequence of "strit" // looking for winner
+    int power_of_pair_carts[numbers_of_player][4];//power of cart that make sets of pair or triple
+    int highest_color_cart[numbers_of_player];//highest card in color set/ if therw is not any color - 0
+    for (int i = 0; i < numbers_of_player; ++i) {
+        for (int j = 0; j < 4; ++j) {
+            power_of_pair_carts[i][j] = 0;
+        }
+    }
+    for (int i = 0; i < numbers_of_player; ++i) {
+        card_t player_cart[7];
+        for (int j = 0; j < 5; ++j) {
+            player_cart[j] = table_carts[j];
+        }
+        player_cart[5] = players[i].cards[0];
+        player_cart[6] = players[i].cards[1];
+        buble_sort_for_carts(player_cart);
+        highest_cart_value[i] = player_cart[0].value;
+        int is_sequence = is_in_sequence(player_cart, &start_of_sequence[i]);
+        int is_color = serching_for_color(player_cart, &highest_color_cart[i]);
+        int strength_of_pair = fstrength_of_pair(player_cart, &power_of_pair_carts[i][0]);
+        if (is_color && is_sequence) {
+            player_score[i] = 8;
+            continue;
+        }
+        if (is_color) {
+            if (strength_of_pair < 4) {
+                player_score[i] = 5;
+                continue;
+            }
+
+            player_score[i] = strength_of_pair + 2;
+            continue;
+        }
+        if (is_sequence) { player_score[i] = 4; } else { player_score[i] = strength_of_pair; }
+    }
+    int count = 0;
+    int how_many_player_with_this_same_amount_of_points = 0;
+    int max = 0;
+    for (int i = 0; i < numbers_of_player ; ++i) {
+        if (player_score[i] > max) { max = player_score[i];continue; }
+        if (player_score[i] == max) {
+            count += 1;
+            if (how_many_player_with_this_same_amount_of_points < count) { how_many_player_with_this_same_amount_of_points = count; }
+        } else count = 0;
+    }
+    if (how_many_player_with_this_same_amount_of_points == 0) {
+        for (int i = 0; i < numbers_of_player; ++i) {
+            if (player_score[i] == max) {
+                table_of_winners[0] = players[i];
+                how_many_players_win[0]+=1;
+                return;
+            }
+        }
+    }
+    int max_value = 0;
+    short par = 0;
+    if (max == 5) {
+        for (int x = 0; x < numbers_of_player; x++) {
+            if (player_score[x] == max && highest_color_cart[x] > max_value) { max_value = highest_color_cart[x];}
+        }
+        for (int i = 0; i < numbers_of_player; ++i) {
+            if (highest_color_cart[i] == max_value && player_score[i] == max) {
+                table_of_winners[par] = players[i];
+                par += 1;
+                how_many_players_win[0]+=1;
```

Comment:
Again: `how_many_players_win` is not an array use `*how_many_players_win`.
And why use par if you already have `how_many_players_win`?

It also applies to rest of the function.